pkg/base/store_spec.go

```diff
@@ -40,17 +40,37 @@ import (
 // hard coded to 640MiB.
 const MinimumStoreSize = 10 * 64 << 20
 
+// GetAbsoluteStorePath takes a (possibly relative) and returns the absolute path.
+// Returns an error if the path begins with '~' or Abs fails.
+// 'fieldName' is used in error strings.
+func GetAbsoluteStorePath(fieldName string, p string) (string, error) {
+	if p[0] == '~' {
+		return "", fmt.Errorf("%s cannot start with '~': %s", fieldName, p)
+	}
+
+	ret, err := filepath.Abs(p)
+	if err != nil {
+		return "", errors.Wrapf(err, "could not find absolute path for %s %s", fieldName, p)
+	}
+	return ret, nil
+}
+
 // StoreSpec contains the details that can be specified in the cli pertaining
 // to the --store flag.
 type StoreSpec struct {
 	Path string
 	// SizeInBytes is used for calculating free space and making rebalancing
 	// decisions. Zero indicates that there is no maximum size. This value is not
 	// actually used by the engine and thus not enforced.
-	SizeInBytes int64
-	SizePercent float64
-	InMemory    bool
-	Attributes  roachpb.Attributes
+	SizeInBytes     int64
+	SizePercent     float64
+	InMemory        bool
+	Attributes      roachpb.Attributes
+	UseSwitchingEnv bool
+	// EncryptionSpec is non-nil after MatchStoreAndEncryptionSpecs if there is an encryption
+	// spec with matching path.
+	//	EncryptionSpec *StoreEncryptionSpec
+	ExtraFields map[string]string
```

Comment:
btw: this clearly needs to go, it's not like we can easily pass a string->string map to cgo, and even if we could, that's a bit fragile.
I'm still trying to figure out if I can have some blind `CCLOptions` of some type, but it's a bit tricky.